tests/x11regressions/firefox/firefox_downloading.pm

```diff
@@ -36,6 +36,7 @@ sub dl_location_switch {
 }
 
 sub dl_save {
+    my ($self) = @_;
     my ($link) = @_;
```

Comment:
Not sure if you need self at all here, as you don't use self in the method. If you want to have both write something like:
my ($self, $link)= @_; This will expand two variable. But method has to be called either like self->dl_save($link) or dl_save($self, $link).